apps/builder/app/builder/features/style-panel/sections/box-shadows/utils.ts

```diff
@@ -0,0 +1,94 @@
+import type {
+  LayerValueItem,
+  LayersValue,
+  StyleProperty,
+  TupleValue,
+  UnparsedValue,
+} from "@webstudio-is/css-data";
+import type { RenderCategoryProps } from "../../style-sections";
+
+export const property: StyleProperty = "boxShadow";
+
+const isValidBoxShadowValue = (layer: LayerValueItem) =>
+  layer.type === "tuple" || layer.type === "unparsed";
+
+export const deleteLayer = (
+  index: number,
+  layers: LayersValue,
+  createBatchUpdate: RenderCategoryProps["createBatchUpdate"]
+) => {
+  const batch = createBatchUpdate();
+  const layer = layers.value[index];
+
+  if (isValidBoxShadowValue(layer) === false) {
+    return;
+  }
+  const newLayers = [...layers.value];
+  newLayers.splice(index, 1);
+
+  if (newLayers.length === 0) {
+    batch.deleteProperty(property);
+  } else {
+    batch.setProperty(property)({
+      type: "layers",
+      value: newLayers,
+    });
+  }
+
+  batch.publish();
+};
+
+export const hideLayer = (
+  index: number,
+  layers: LayersValue,
+  createBatchUpdate: RenderCategoryProps["createBatchUpdate"]
+) => {
+  const batch = createBatchUpdate();
+  const layer = layers.value[index];
+
+  if (isValidBoxShadowValue(layer) === false) {
+    return;
+  }
+  const newLayers = [...layers.value];
+  newLayers.splice(index, 1, {
+    ...(layer as TupleValue | UnparsedValue),
```

Comment:
Type casting is needed here. It is not picking up the types if i use `isValidBoxShadowValue() === false` syntax.

Reply:
You can just write checks without isValidBoxShadowValue. There is no big logic in it.